### ISSUE TITLE
@connection directive support

### ISSUE THREAD

**Issue by bennyhobart** [convoyinc/apollo-cache-hermes#289]
Hi guys, I'm not sure if this is an issue with my application but it seems like the @connection directive isn't working as expected

the two places I use it are as such, (code stripped for brevity)
```
activities(first: 30) @connection(key: "activities")
activities(first: 30, after: $cursor) @connection(key: "activities") 
```

If I remove the cursor from the second query it is working fine, any ideas?

**Comment by nevir** [comment]
Ah, hmm, this cache doesn't have explicit support for `@connection`—though, IIRC, most of that is handled by Apollo client explicitly.

I'm curious: what keys end up being written to when you load the first query as opposed to the second?  Try turning on [verbose mode](https://github.com/convoyinc/apollo-cache-hermes/blob/master/src/context/CacheContext.ts#L127-L135), which'll log out snapshots for each write and lists of ids that have changed; might help us narrow it down

**Comment by bennyhobart** [comment]
@nevirI picked this out from the logs, it looks like they are being stored in two different locations

this is the root log ![screen shot 2018-01-16 at 2 06 42 pm](https://user-images.githubusercontent.com/3694930/34970101-88f26916-fac6-11e7-9ab4-285a50cb3d07.png)

this is the important part ![screen shot 2018-01-16 at 2 06 08 pm](https://user-images.githubusercontent.com/3694930/34970111-94288630-fac6-11e7-9288-b56cc84b8ebb.png)

Any action I can take to help

**Comment by bennyhobart** [comment]
I think at this point (https://github.com/convoyinc/apollo-cache-hermes/blob/5e2b7604fec684cd27ffd9a585e706a704415c4e/src/operations/SnapshotEditor.ts#L279) we determine if the field is a connection directive? and change the key based on this

**Comment by nevir** [comment]
Yeah, or potentially earlier when parsing the query ([`ParsedQueryNode`](https://github.com/convoyinc/apollo-cache-hermes/blob/master/src/ParsedQueryNode.ts)) - however, we also have to teach the cache how to combine the two lists too (maybe append by default, but that quickly falls over), right?

**Comment by nevir** [comment]
Re:

> Any action I can take to help?

We're unfortunately not going to have much bandwidth to dig into this much for the next couple weeks—if you want to take a stab at an implementation, that'd be a huge help!  It might also be worth checking out how [inmemory](https://github.com/apollographql/apollo-client/tree/master/packages/apollo-cache-inmemory) handles the `@connection` directive

**Comment by bennyhobart** [comment]
I'll take a stab when I get the opportunity, we should avoid merging there isn't enough information to do so, overwrite the previous result with the new which will keep inserts fast, it's the responsibility of the application developer — when (fetching more)[https://www.apollographql.com/docs/react/recipes/pagination.html#fetch-more]) — to maintain this state, thanks for the research into in-memory cache's implementation
